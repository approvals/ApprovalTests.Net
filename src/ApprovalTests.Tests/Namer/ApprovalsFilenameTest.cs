public class ApprovalsFilenameTest
{
    [Test]
    public void TestMachineSpecificName()
    {
        // begin-snippet: approvals_filename
        var approvalsFilename = ApprovalsFilename.Parse(@"..\Email\EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml");
        // end-snippet
        Approvals.Verify(approvalsFilename);
        ClassicAssert.True(approvalsFilename.IsMachineSpecific);
    }

    [Test]
    public void TestNonMachineSpecificName() =>
        Approvals.Verify(ApprovalsFilename.Parse(@"..\Email\EmailTest.Testname.approved.eml"));

    [Test]
    public void TestSimilarFiles()
    {
        var file = PathUtilities.GetAdjacentFile(@"..\Email\EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml");
        var approvalsFilename = ApprovalsFilename.Parse(file);
        Approvals.VerifyAll("Like EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml", approvalsFilename.GetOtherMachineSpecificFiles(), f => f.Name);
    }
}