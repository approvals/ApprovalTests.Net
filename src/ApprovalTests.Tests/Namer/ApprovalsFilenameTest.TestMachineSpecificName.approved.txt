GetFullPath: ..\Email\EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml
Directory: ..\Email
ClassName: EmailTest
MethodName: Testname
AdditionalInformation: [Microsoft_Windows_10_Education]
ApprovedStatus: approved
Extension: eml